internal/filesystem/filesystem.go

```diff
@@ -1,31 +1,164 @@
 package filesystem
 
 import (
+	"context"
+	"errors"
+	"fmt"
 	"io"
 	"io/fs"
+	"os"
+	"time"
 )
 
-// TODO: name?
-type FS interface {
+type (
+	// TODO: [review] These interfaces come from our pre fs.FS implementations
+	// Things like names and patterns may (need to) change.
+	// Some things might need to be added as well.
+	IDFS interface {
+		fs.FS
+		ID() ID
+	}
+	// TODO: better name; OpenPOSIXFile?
+	OpenFileFS interface {
+		fs.FS
+		OpenFile(name string, flag int, perm fs.FileMode) (fs.File, error)
+	}
+	MakeFileFS interface {
+		fs.FS
+		MakeFile(string) error
+	}
```

Comment:
We should try to mimic the Go standard `os` pkg as close as possible, unless we really need to deviate. With exceptions being to target `fs` pkg standards rather than those from `os`.
E.g. this should probably become
```suggestion
	CreateFileFS interface {
		fs.FS
		CreateFile(string) (fs.File, error)
	}
```
to match `os.Create(name string) (*os.File, error)`

Reply:
Done in f9a75020973f3af2b73f2889cb025e6c094904a8